src/compiler/crystal/syntax/to_s.cr

```diff
@@ -202,11 +202,61 @@ module Crystal
 
     def visit(node : NamedTupleLiteral)
       @str << '{'
-      node.entries.join(@str, ", ") do |entry|
+
+      # short-circuit to handle empty named tuple context
+      if node.entries.empty?
+        @str << '}'
+        return false
+      end
```

Comment:
There's at least one case somewhere where an empty named tuple literal is to be stringified. Having this early return prevents having to use `.first?` and `.last?` and dealing with `nil`.